presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -158,14 +163,32 @@ private void validateShowStatsSubquery(ShowStats node, Query query, QuerySpecifi
             check(!querySpecification.getGroupBy().isPresent(), node, "GROUP BY is not supported in SHOW STATS SELECT clause");
             check(!querySpecification.getSelect().isDistinct(), node, "DISTINCT is not supported by SHOW STATS SELECT clause");
 
+            ImmutableSet.Builder<String> columnsBuilder = ImmutableSet.builder();
+            boolean allColumns = false;
             List<SelectItem> selectItems = querySpecification.getSelect().getSelectItems();
-            check(selectItems.size() == 1 && selectItems.get(0) instanceof AllColumns, node, "Only SELECT * is supported in SHOW STATS SELECT clause");
+            for (SelectItem selectItem : selectItems) {
+                if (selectItem instanceof AllColumns) {
+                    allColumns = true;
+                    continue;
```

Comment:
'break'?

Reply:
@arhimondr I'm using continue to make sure I check all the selectItems, e.g. I want the following query to fail:

`SHOW STATS FOR (SELECT *, nationkey + 1 FROM nation)`